src/commands.cc

```diff
@@ -2556,19 +2557,34 @@ const CommandDesc declare_user_mode_cmd = {
     "declare-user-mode",
     nullptr,
     "declare-user-mode <name>: add a new user keymap mode",
-    single_param,
+    ParameterDesc{
+        { { "docstring", { true, "user mode description" } } },
+        ParameterDesc::Flags::SwitchesOnlyAtStart, 1, 2
+    },
     CommandFlags::None,
     CommandHelper{},
     CommandCompleter{},
     [](const ParametersParser& parser, Context& context, const ShellContext&)
     {
-        context.keymaps().add_user_mode(std::move(parser[0]));
+        auto docstring = parser.get_switch("docstring").value_or(StringView{});
+        context.keymaps().add_user_mode(std::move(parser[0]), std::move(docstring.str()));
```

Comment:
That second std::move is unnecessary, str() already returns a temporary so it will be moved by default.